pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/route/GetParticipantFileDownloadRoute.java

```diff
@@ -0,0 +1,91 @@
+import org.broadinstitute.ddp.db.dao.AnswerDao;
+import org.broadinstitute.ddp.db.dao.UserDao;
+import org.broadinstitute.ddp.db.dto.ActivityInstanceDto;
+import org.broadinstitute.ddp.json.FileDownloadResponse;
+import org.broadinstitute.ddp.json.errors.ApiError;
+import org.broadinstitute.ddp.model.activity.instance.answer.Answer;
+import org.broadinstitute.ddp.model.user.User;
+import org.broadinstitute.ddp.service.FileDownloadService;
+import org.broadinstitute.ddp.util.ResponseUtil;
+import spark.Request;
+import spark.Response;
+import spark.Route;
+
+import java.io.FileNotFoundException;
+import java.net.URL;
+import java.util.Optional;
+
+@Slf4j
+@AllArgsConstructor
+public class GetParticipantFileDownloadRoute implements Route {
+    private static FileDownloadService fileDownloadService;
+
+    public GetParticipantFileDownloadRoute(FileDownloadService service) {
+        fileDownloadService = service;
+    }
+
+    @Override
+    public Object handle(Request request, Response response) {
+        String studyGuid = request.params(RouteConstants.PathParam.STUDY_GUID);
+        String participantGuid = request.params(RouteConstants.PathParam.USER_GUID);
+        String instanceGuid = request.params(RouteConstants.PathParam.INSTANCE_GUID);
+        String questionStableId = request.params(RouteConstants.PathParam.QUESTION_STABLE_ID);
+
+        log.info("Fetching signed url for file download of activity instance {} and participant {} in study {} for Question: {})",
+                instanceGuid, participantGuid, studyGuid, questionStableId);
+
+        if (StringUtils.isEmpty(studyGuid) || StringUtils.isEmpty(participantGuid)
+                || StringUtils.isEmpty(instanceGuid) || StringUtils.isEmpty(questionStableId)) {
+            throw ResponseUtil.haltError(response, 400, new ApiError(ErrorCodes.BAD_PAYLOAD, "Required parameter(s) missing"));
+        }
+
+        return TransactionWrapper.withTxn(handle -> {
+            //validate user
+            UserDao userDao = handle.attach(UserDao.class);
+            Optional<User> userOpt = userDao.findUserByGuid(participantGuid);
+            if (!userOpt.isPresent()) {
+                throw ResponseUtil.haltError(response, 404, new ApiError(ErrorCodes.USER_NOT_FOUND, "User not found"));
+            }
+            Long userId = userOpt.get().getId();
+
+            ActivityInstanceDao instanceDao = handle.attach(ActivityInstanceDao.class);
+            Optional<ActivityInstanceDto> instanceDtoOpt = instanceDao.findByActivityInstanceGuid(instanceGuid);
+            if (!instanceDtoOpt.isPresent()) {
+                throw ResponseUtil.haltError(response, 404, new ApiError(ErrorCodes.ACTIVITY_NOT_FOUND, "Instance not found"));
+            }
+            //make sure the instanceGuid passed belong to the participant guid passed
+            ActivityInstanceDto activityInstanceDto = instanceDtoOpt.get();
+            if (activityInstanceDto.getParticipantId() != userId) {
```

Comment:
Will this hold true for proxies?

Reply:
activityInstance always belong to participant, so this should be good for proxies too.
Proxy value comes as operatorGuid and UserAuthCheckFilter checks and makes sure operator has access to this participant data. UserAuthCheckFilter/DDPAuth/UserPermissions.canAccessStudyDataForUser.

Reply:
I'm still missing something.  Where is the currently authenticated user checked to see if they are the participant or a proxy for them?  I see the parameter extracted from the URL path, but not where we're checking the value from the Bearer auth.